pkgs/os-specific/linux/kernel/manual-config.nix

```
@@ -132,6 +132,11 @@ let
             --replace NIXOS_RANDSTRUCT_SEED \
             $(echo ${randstructSeed}${src} ${configfile} | sha256sum | cut -d ' ' -f 1 | tr -d '\n')
         fi
+        if [ -f scripts/gen-randstruct-seed.sh ]; then
+          substituteInPlace scripts/gen-randstruct-seed.sh \
+            --replace NIXOS_RANDSTRUCT_SEED \
+            $(echo ${randstructSeed}${src} ${configfile} | sha256sum | cut -d ' ' -f 1 | tr -d '\n')
+        fi
```

Instead of duplicating the substituteInPlace, iterate over the files that need this treatment.

(Note: Due to GitHub bug https://github.com/isaacs/github/issues/284 / https://github.com/community/community/discussions/4452 , I am unable to correctly set the range-to-be-replaced of this suggested change.  It should include two additional lines above the range GitHub displays.)

```suggestion
        for file in \
          scripts/gcc-plugins/gen-random-seed.sh \
          scripts/gen-randstruct-seed.sh \
        ;do
          if [ -f "$file" ]; then
            substituteInPlace "$file" \
              --replace NIXOS_RANDSTRUCT_SEED \
              $(echo ${randstructSeed}${src} ${configfile} | sha256sum | cut -d ' ' -f 1 | tr -d '\n')
          fi
        done
```

---

Good idea!  I applied your suggestion and confirmed I can still rebuild my kernel.